bundler/spec/dependabot/bundler/file_updater_spec.rb

```diff
@@ -1443,6 +1443,33 @@
       end
     end
 
+    context "for a gem that depends on bundler" do
+      subject(:updated_gemfile) do
+        updated_files.find { |f| f.name == "Gemfile" }
+      end
+
+      let(:dependency_files) { bundler_project_dependency_files("guard_bundler") }
+      let(:dependency_name) { "guard-bundler" }
+      let(:dependency_version) { "3.0.0" }
+      let(:dependency_previous_version) { "2.2.1" }
+      let(:requirements) do
+        [{
+          file: "Gemfile",
+          requirement: "~> 2.2.1",
+          groups: [],
+          source: nil
+        }]
+      end
+
+      it "raises an error", :bundler_v1_only do
```

Comment:
I don't think there's a way to trigger this error anymore, it'll just not propose an update I think

Reply:
That was my thought as well, but turns out the UpdateChecker isn't currently smart enough, demonstrated here: https://github.com/dependabot/dependabot-core/pull/3418